spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -419,5 +421,13 @@ public void testFaultToleranceOnWrite() throws IOException {
 
     assertThat(snapshotBeforeFailingWrite).isEqualTo(snapshotAfterFailingWrite);
     assertThat(resultBeforeFailingWrite).isEqualTo(resultAfterFailingWrite);
+
+    while (location.exists()) {
```

Comment:
this seems like a workaround. https://stackoverflow.com/questions/56290320/junit-cannot-delete-tempdir-with-file-created-by-spark-structured-streaming indicates that the issue might be due to a resource not being closed, so maybe we should investigate what the actual root cause of this issue is